### PR TITLE
Added an example to demonstrate ideal field weakening 

### PR DESCRIPTION
**Pull request by AHaumer**
of a permanent magnet synchronous machine.

### CODE REVIEW

File: Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/SynchronousMachines/SMPM_FieldWeakening.mo [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  connect(terminalBox.starpoint, groundM.pin)
+    annotation (Line(points={{30,-14},{20,-14},{20,-20}}, color={85,170,255}));
+  connect(referenceCurrentSource.plug_n, resistorS.plug_n)
+    annotation (Line(points={{40,30},{60,30}}, color={85,170,255}));
+  annotation (
+    experiment(
+      StopTime=2.5,
+      Interval=0.0001,
+      Tolerance=1e-06,
+      __Dymola_Algorithm="Dassl"),
```
Comment (beutlich) [inline Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/SynchronousMachines/SMPM_FieldWeakening.mo:156]: Solver annotation should not be set.

**Review by christiankral — COMMENTED** [review]
OK, I like the example and yet have some comments for which I will also prepare some code updates after having some discussions:

image

- As the speed is prescribed by a ramp it makes sense to set `Jr = 0` to avoid inertia effects when investigating `smpm.powerBalance.powerMechanical`
- On my understanding the *nominal* current is the current to not exceed the maximum operating temperature in steady state operation
- On my understanding the *maximum* current is related with the overload operation of the machine which in steady state would exceed the maximum operation temperature; however, the related inverter has to be capable of providing the required maximum current 
    - The drive runs to up to four times the nominal speed: what application could this be? To me such a speed range were typical for an electric vehicle
    - As soon as the speed exceeds the nominal speed the current has to exceed the *nominal* current to maintain the desired (nominal?) torque
    - In case of an electric vehicle it shall be capable to exceed the nominal speed in steady state but this is conflict with my understanding of what *nominal* (current) means  

In think the documentation of the example shall be enhanced (after discussions):
- Explain the terms *nominal* and *maximum* current
- Explain whether and what is *nominal* torque and overload operation
- Explain what result variables shall be plotted and what they show

**Comment by christiankral** [comment]
When increasing the current limit from `IqNominal = 100 A` to `IqNominal = 150 A` the overall torque can be increase 

image

without exceeding the nominal current:

image

Wouldn't that be the preferred behavior for an electric vehicle application?

File: Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/SynchronousMachines/SMPM_FieldWeakening.mo [hunk 2]
```diff
@@ -0,0 +1,185 @@
+within Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.SynchronousMachines;
+model SMPM_FieldWeakening
+  "Test example: Investigate field weakening of a PermanentMagnetSynchronousMachine"
+  extends Modelica.Icons.Example;
+  import Modelica.Constants.pi;
+  constant Integer m=3 "Number of phases";
+  parameter SI.Voltage VNominal=100 "Nominal RMS voltage per phase";
+  parameter SI.Current IqNominal=100 "Nominal RMS q current per phase";
```
Comment (christiankral) [inline Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/SynchronousMachines/SMPM_FieldWeakening.mo:8]:
```suggestion
  parameter SI.Current IqNominal=150 "Nominal RMS q current per phase";
```

**Review by christiankral — COMMENTED** [review]
Suggestion

**Review by christiankral — APPROVED** [review]
Looks good now

**Pull request merged**